docs(linux): dedupe installation instructions and update conf desc (#178)
logdna/logdna-agent-v2: docs/LINUX.md

# LogDNA Agent on Linux distributions

> The LogDNA Agent for Linux collects log data from your Linux environment. The agent is available as a Linux package for Debian-based distributions using `apt` and Red Hat-based distributions using `yum`. If you use a different package manager that does not support `.deb` or `.rpm` files, you can still use the agent by [manually compiling the binary](https://github.com/logdna/logdna-agent-v2/blob/eb06d4f3f7c1033b494f1f0439957f96533f9225/docs/README.md#building-agent-binary-on-linux).

Reviewer: NIT: I think Red Hat-based distribution should be replaced with RPM-based distributions, as a way to note "Linux distributions that use the RPM Package Manager".

https://en.wikipedia.org/wiki/Category:RPM-based_Linux_distributions

Using Red Hat distros in this context leaves SUSE family out :)

> ## Table of Contents

Reviewer: To simplify reading for new users, I think the order should be:

1. Installing
2. Usage
3. Upgrading from legacy Linux Agent

That way, new users can read the required steps in order (1 & 2) without jumping through the page.

Reviewer 2: @jorgebay I do see your point, but after a ton of conversation we decided to separate the install from upgrade (as you originally had it); then I decided to move the usage section come last since that was accurate for all audiences.

Reviewer: ok, sounds good.

  * [Installation: fresh installs](#Installation)

>   * [Upgrading: migrating from legacy Linux agent](#upgrade-migration)

Reviewer: Note that the bookmark is broken, it should be `#upgrademigration`

  * [Usage](#usage)
    * [Enable the agent](#enable-the-agent)
    * [Configure the agent](#configure-the-agent)
    * [Run the agent](#run-the-agent)

## Installation

1. To download the agent and resources package, open a host terminal and run the appropriate commands based on the Linux distribution.

> * Debian-based distributions

Reviewer 3: style-wise, with the bullets, it's harder to read `Debian-based distributions` and `RPM/Red Hat-based distributions`. Can we bold it instead?

```bash
echo "deb https://assets.logdna.com stable main" | sudo tee /etc/apt/sources.list.d/logdna.list
wget -qO - https://assets.logdna.com/logdna.gpg | sudo apt-key add -
sudo apt-get update
```
* **RPM/Red Hat-based distributions**
```bash
sudo rpm --import https://assets.logdna.com/logdna.gpg
echo "[logdna]
name=LogDNA packages
baseurl=https://assets.logdna.com/el6/
enabled=1
gpgcheck=1
gpgkey=https://assets.logdna.com/logdna.gpg" | sudo tee /etc/yum.repos.d/logdna.repo
```

> 2. To upgrade the agent, use the commands below, based on the distro:

Reviewer: ```suggestion
2. To install the agent, use the commands below based on the Linux distribution:
```

> * Debian-base distributions:

Reviewer 3: ```suggestion
* Debian-based distributions:
```

```bash
sudo apt-get install logdna-agent
```
* **RPM/Red Hat-based distributions:**
```bash

> sudo yum -y install logdna-agent

Reviewer: We should remove the auto yes flag here.

```

## Upgrade/migration

> └ **Note** [for users migrating from the [legacy LogDNA Agent](https://github.com/logdna/logdna-agent)]:

Reviewer 2: The format you suggest is in our other docs, so yes, let's stick with it.

> If you have previously installed the `logdna-agent` Linux package and have an existing `/etc/logdna.conf` file, it is _still_ recommended to follow the instructions below to ensure that all users are retrieving the latest version of the package from the correct source repository.

1.  To download the agent and resources package, open a host terminal and run the appropriate commands based on the Linux distribution.

> * Debian-base distributions:

Reviewer 3: ```suggestion
* Debian-based distributions:
```

```bash
echo "deb https://assets.logdna.com stable main" | sudo tee /etc/apt/sources.list.d/logdna.list
wget -qO - https://assets.logdna.com/logdna.gpg | sudo apt-key add -
sudo apt-get update
```

* RPM/Red Hat-based distributions:
```bash
sudo rpm --import https://assets.logdna.com/logdna.gpg
echo "[logdna]
name=LogDNA packages
baseurl=https://assets.logdna.com/el6/
enabled=1
gpgcheck=1
gpgkey=https://assets.logdna.com/logdna.gpg" | sudo tee /etc/yum.repos.d/logdna.repo
```

2. To upgrade the agent, use the commands below, based on the distro:

> * Debian-base distributions:

Reviewer 3: ```suggestion
* Debian-based distributions:
```

```bash
sudo apt-get upgrade logdna-agent
```
* **RPM/Red Hat-based distributions:**
```bash
sudo yum update logdna-agent
```

## Usage
The agent uses [**systemd**](https://systemd.io/) to run as a Linux daemon. The installed package provides a **systemd** unit file for the `logdna-agent` service, which is defined to execute a daemon process with the compiled agent binary. The process that is started in the service is managed by **systemd** and can be interfaced with using `systemctl`.

### _Enable the agent_

1.  Reload the **systemd** unit files to obtain the most recent version of the agent:
```bash
    sudo systemctl daemon-reload
```

2.  Enable the `logdna-agent` service using the `systemctl` command:
```bash
    sudo systemctl enable logdna-agent
```

### _Configure the agent_

└ Note \[for users upgrading /migrating from the [legacy LogDNA

> Agent](https://github.com/logdna/logdna-agent)\]: Many users will likely already have a configuration file `/etc/logdna.conf` from prior installations. The LogDNA Agent 3.3 does not utilize the legacy `/etc/logdna.conf` file, but instead uses a **systemd** unit file `/etc/logdna.env`.

Reviewer 2: Ah yes, I shouldn't have said it isn't used, for sure.

However, I *think*we came to the decision that everyone, even people upgrading, should still create the `logdna.env` file, for consistency's sake. So the following steps apply to everyone.  But we can certainly change it again if we want. ;)

gpgcheck=1
gpgkey=https://assets.logdna.com/logdna.gpg" | sudo tee /etc/yum.repos.d/logdna.repo

# install the package via yum (skip this step if already installed)
sudo yum -y install logdna-agent

# update the package via yum
sudo yum update logdna-agent
```
#
## Usage
The agent uses [systemd](https://systemd.io/) to run as a Linux daemon. The installed package provides a `systemd` unit file for the `logdna-agent` service which is defined to execute a daemon process with the compiled agent binary. The process started in the service is managed by `systemd` and can be interfaced with using `systemctl`.

---
### _Enabling the agent_

> - Reload the `systemd` unit files to ensure the most recent version of the agent is used & enable the service via `systemctl`:

Author: Instead of replacing the ampersand & with "and to", I moved the latter part into its own bullet since I think there's a slight distinction to be made. Reloading the files isn't a requirement in enabling the service, they are 2 separate commands so I broke them up accordingly.

  ```bash
  sudo systemctl daemon-reload
  sudo systemctl enable logdna-agent
  ``` 
---
### _Configuring the agent_

> └ **Note** [for users upgrading from older versions of [logdna-agent-v2](https://github.com/logdna/logdna-agent-v2) or migrating from the legacy [logdna-agent](https://github.com/logdna/logdna-agent)]: 

Author: I think using the "proper" names here might make it even more confusing. The "LogDNA Agent v2" project name already includes a version but the project itself is also versioned; throwing in the legacy agent and naming it "LogDNA Agent v1" seems like it would only cause more trouble

> └ **Note** [for users upgrading from older versions of [logdna-agent-v2](https://github.com/logdna/logdna-agent-v2) or migrating from the legacy [logdna-agent](https://github.com/logdna/logdna-agent)]: 

Reviewer 2: Agree, let's not use the v1 as I suggested above.

  > Many users will likely already have a configuration file `/etc/logdna.conf` from prior setup(s). While the config variable names have been changed / updated between [logdna-agent](https://github.com/logdna/logdna-agent) and [logdna-agent-v2](https://github.com/logdna/logdna-agent-v2), backwards compatibility has been maintained so older `.conf` files should still be valid. If this is applicable to you and there are no changes that need to be made to your config, you may be able to skip this section and try to start the service using your existing configuration
#

- By default, `/etc/logdna.conf` is the expected path for the agent's configuration file. Make sure to create this file if it doesn't exist since it's required to start the agent.

    ```bash
    sudo touch /etc/logdna.conf # it can initalized as an empty file, you will be adding to it in the steps below
    ```

>     This file stores key-value pairs as environment variables which can be injected into the agent at runtime and manage a variety of configuration options. You can see all the available variable options via `logdna-agent --help` or refer to them [here](README.md#options). If you're coming from the legacy agent, take note of the variables which have been changed, updated, and deprecated when compared to [logdna-agent](https://github.com/logdna/logdna-agent).

Author: In this case, I think we'd want to reference the legacy project for comparison since these docs would imply they're already looking at the `logdna-agent-v2` project.

- Before starting the `logdna-agent` service, an ingestion key must be provided by setting the `LOGDNA_INGESTION_KEY` variable in `/etc/logdna.conf`:
  
  ```edit /etc/logdna.conf
  LOGDNA_INGESTION_KEY=<YOUR INGESTION KEY HERE>
  ```
  The ingestion key is the only required variable, but users can set additional variables in order to meet their desired configuration. For example, to attach a "production" tag to every log line - you could set the `LOGDNA_TAGS` variable in `/etc/logdna.conf`:

> 

Reviewer 2: replace "users" with "you."

replace hyphen with a comma

gpgcheck=1
gpgkey=https://assets.logdna.com/logdna.gpg" | sudo tee /etc/yum.repos.d/logdna.repo

# install the package via yum (skip this step if already installed)
sudo yum -y install logdna-agent

# update the package via yum
sudo yum update logdna-agent
```
#
## Usage
The agent uses [systemd](https://systemd.io/) to run as a Linux daemon. The installed package provides a `systemd` unit file for the `logdna-agent` service, which is defined to execute a daemon process with the compiled agent binary. The process that is started in the service is managed by `systemd` and can be interfaced with using `systemctl`.

---
### _Enabling the agent_

> - Reload the `systemd` unit files to ensure that the most recent version of the agent is used

Reviewer 2: Let's rework this section so that it looks like this:

- Reload the `systemd` unit files to ensure that the most recent version of the agent is used.

  ```bash
  sudo systemctl daemon-reload

- Enable the `logdna-agent` service using the `systemctl` command.

  ```bash
  sudo systemctl enable logdna-agent

  ```bash
  sudo systemctl daemon-reload
  sudo systemctl enable logdna-agent
  ``` 
---
### _Configuring the agent_
└ **Note** [for users upgrading from older versions of the [LogDNA Agent](https://github.com/logdna/logdna-agent-v2) or migrating from the [Legacy LogDNA Agent](https://github.com/logdna/logdna-agent)]: 
  > Many users will likely already have a configuration file `/etc/logdna.conf` from prior installations. While the config variable names have been changed / updated between the [Legacy LogDNA Agent](https://github.com/logdna/logdna-agent) and the current [LogDNA Agent](https://github.com/logdna/logdna-agent-v2), backwards compatibility has been maintained so older `.conf` files are still valid. If this is applicable to you and there are no changes that need to be made, you can skip this section and start the service using your existing configuration.
#

- By default, `/etc/logdna.conf` is the expected path for the agent's configuration file. Make sure to create this file if it doesn't exist since it's required to start the agent.

    ```bash
    sudo touch /etc/logdna.conf # it can initalized as an empty file, you will be adding to it in the steps below

>     ```

Reviewer 2: Missing an "i" in "initialized."
